### ISSUE TITLE
`DiscreteJaxOperator.n_conn` wrong result

### ISSUE THREAD

**Issue by PhilipVinc** [netket/netket#2227]
Reported by email by a benefactor.

>   More specifically, I was looking at the default implementation of DiscreteJaxOperator.n_conn() in netket/operator/
 _discrete_operator_jax.py.
> 
>  It currently seems to do:
>
>  _, mels = self.get_conn_padded(x)
> nonzeros = jnp.abs(x) > 0
>  _n_conn = nonzeros.sum(axis=-1)
